File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    new_message = Message.create!(content: data['message']) # 受け取ったメッセージをDBへ保存
    template = ApplicationController.renderer.render(partial: 'messages/message', locals: {message: new_message})
    ActionCable.server.broadcast 'room_channel', template
  end
end

File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

const room_channel = consumer.subscriptions.create("RoomChannel", {
  connected() {
    console.log("connected")
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(text) {
    var messages = document.getElementById('messages')
    messages.innerHTML += text
    // Called when there's incoming data on the websocket for this channel
  },

  speak: function(text) {
    return this.perform('speak', {message: text});
  }
});

document.addEventListener('DOMContentLoaded', function(){
  var input = document.getElementById('chat-input')
  var button = document.getElementById('send-button')
  button.addEventListener('click', function(){
    var text = input.value
    const rc = Object.create(room_channel)
    rc.speak(text)
    input.value = ''
  })
})